tests/manage/z_cluster/nodes/test_worker_nodes_network_failures.py

```diff
@@ -0,0 +1,212 @@
+
+
+@tier4
+@tier4c
+@ignore_leftovers
+class TestWorkerNodesFailure(ManageTest):
+    """
+    Test all worker nodes simultaneous abrupt network failure for ~300 seconds
+    """
+    pvc_size = 5  # size in Gi
+    num_of_app_pods = 2  # Pods of each interface type to be run
+    short_nw_fail_time = 300  # Duration in seconds for short network failure
+
+    @pytest.fixture()
+    def setup(self, request, scenario, nodes, dc_pod_factory):
+        """
+        Identify the nodes and start DeploymentConfig based app pods
+        """
+        worker_nodes = helpers.get_worker_nodes()
+        ocs_nodes = machine.get_labeled_nodes(constants.OPERATOR_NODE_LABEL)
+        non_ocs_nodes = list(set(worker_nodes) - set(ocs_nodes))
+
+        def finalizer():
+            # Restart the nodes if in NotReady state
+            not_ready_nodes = [
+                n for n in node.get_node_objs() if n
+                .ocp.get_resource_status(n.name) == constants.NODE_NOT_READY
+            ]
+            if not_ready_nodes:
+                logger.info(
+                    f"Nodes in NotReady status found: {[n.name for n in not_ready_nodes]}"
+                )
+                nodes.restart_nodes(not_ready_nodes)
+
+            node.wait_for_nodes_status(status=constants.NODE_READY)
+            helpers.remove_label_from_worker_node(
+                node_list=worker_nodes, label_key="nodetype"
+            )
+            ceph_health_check()
+
+        request.addfinalizer(finalizer)
+
+        if (scenario == 'dedicated') and len(non_ocs_nodes) == 0:
+            # TODO: Add required num of nodes instead of skipping
+            # https://github.com/red-hat-storage/ocs-ci/issues/1291
+            pytest.skip("Add node not implemented, github issue #1291")
+
+        app_pod_nodes = ocs_nodes if (scenario == "colocated") else non_ocs_nodes
+
+        ceph_cluster = CephCluster()
+
+        # Label nodes to be able to run app pods
+        helpers.label_worker_node(
+            node_list=app_pod_nodes, label_key="nodetype", label_value="app-pod"
+        )
+
+        # Create deploymentconfig based pods for both interface
+        pod_objs = list()
+        interface_list = [constants.CEPHBLOCKPOOL, constants.CEPHFILESYSTEM]
+        logger.info(f"Starting app pods")
+        for interface in interface_list:
+            for num in range(self.num_of_app_pods):
+                pod_objs.append(
+                    dc_pod_factory(
+                        interface=interface, size=self.pvc_size,
+                        node_selector={'nodetype': 'app-pod'}
+                    )
+                )
+
+        return ceph_cluster, pod_objs, worker_nodes
+
+    @pytest.mark.parametrize(
+        argnames=["scenario"],
+        argvalues=[
+            pytest.param(
+                *['colocated'], marks=pytest.mark.polarion_id("OCS-1432")
+            ),
+            pytest.param(
+                *['dedicated'], marks=pytest.mark.polarion_id("OCS-1433")
```

Comment:
when do we test with dedicated nodes? Does it require a special deployment with extra non OCS nodes to be created as part of cluster deployment?
Asking since I'm pretty sure we don't do that at all in our regression runs and I wouldn't want this test case not to be tested at all

Reply:
Yes, this requires an extra non-OCS node to run, waiting for this [issue 1291](https://github.com/red-hat-storage/ocs-ci/issues/1291) to get resolved.

Reply:
Please check if https://github.com/red-hat-storage/ocs-ci/pull/1661 answers that

Reply:
IIUC, the above said PR implements addition of RHEL worker nodes to AWS, but other functionalities are yet to be implemented.

Infra ticket:
https://projects.engineering.redhat.com/browse/OCSQE-184

Reply:
OK. Please report an ocs-ci issue for tracking this for when the above infra ticket is resolved

Reply:
updated the test for adding the required node.